### PR TITLE
feat: transfer fungible token tool

### PR DESCRIPTION
**Pull request by skurzyp-blockydevs**
**Description**:

**Related issue(s)**:

related to #911 

**Notes for reviewer**:

**Checklist**

- [x] Documented (Code comments, README, etc.)
- [x] Tested (unit, integration, etc.)

### CODE REVIEW

File: packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts [hunk 1]
```diff
@@ -612,6 +614,58 @@ export default class HederaParameterNormaliser {
...
+      client,
+    );
+
+    const tokenInfo = await mirrornode.getTokenInfo(parsedParams.tokenId);
+    const tokenDecimals = Number(tokenInfo.decimals);
+
+    const tokenTransfers: TokenTransferMinimalParams[] = [];
+    let totalAmount = 0;
+
+    for (const transfer of parsedParams.transfers) {
```
Comment (MWBlocky) [inline packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts:646]: ### Token transfers are not guaranteed to net to zero (transaction rejected on-chain)

In `normaliseTransferFungibleToken`, the sender's debit is computed **independently** from the recipients' credits:

```ts
let totalAmount = 0;
for (const transfer of parsedParams.transfers) {
  totalAmount += transfer.amount;                                  // sum in JS float, display units
  tokenTransfers.push({ ..., amount: toBaseUnit(transfer.amount, tokenDecimals).toNumber() });
}
tokenTransfers.push({ ..., accountId: senderAccountId,
  amount: toBaseUnit(-totalAmount, tokenDecimals).toNumber() });   // line 653 — floored separately
```

Hedera requires the token transfer list to sum to exactly zero per token; a non-zero sum is rejected at precheck (`INVALID_ACCOUNT_AMOUNTS`). Because the debit is derived by re-summing the display-unit amounts in JS floating point and flooring that total **separately** from the per-recipient floors, the credits and the debit do not always cancel out.

**Reproduced (real `toBaseUnit`, not mocked):**

| Input | Resulting base units | Sum |
|---|---|---|
| `0.1` + `0.2` @ 8 decimals | `[10000000, 20000000, -30000001]` | **−1** ❌ |
| `1.5` @ 0 decimals | `[1, -2]` | **−1** ❌ |
| `0.001` ×2 @ 2 decimals | `[0, 0, -1]` | **−1** ❌ |

**Root cause:** two combined effects — (1) floating-point accumulation, e.g. `0.1 + 0.2 === 0.30000000000000004`, and (2) floor asymmetry, `floor(-x) !== -floor(x)` for non-integer `x`. `0.1`/`0.2` to two recipients on an 8-decimal token is an ordinary transfer, so this hits the common path.

**Why tests don't catch it:** `transfer-fungible-token-params-normalization.unit.test.ts` mocks `toBaseUnit` as `amount * 10 ** decimals` (no flooring) and only uses integer amounts, so the imbalance can never appear.

**Suggested fix** — derive the sender's debit from the sum of the already-converted base-unit credits, so the list nets to zero by construction:

```ts
const safeDecimals = Number.isFinite(tokenDecimals) ? tokenDecimals : 0;
let totalBase = 0;
for (const transfer of parsedParams.transfers) {
  const base = toBaseUnit(transfer.amount, safeDecimals).toNumber();
  totalBase += base;
  tokenTransfers.push({ tokenId: parsedParams.tokenId, accountId: transfer.accountId, amount: base });
}
tokenTransfers.push({ tokenId: parsedParams.tokenId, accountId: senderAccountId, amount: -totalBase });
```

Please also add a regression test that uses the **real** `toBaseUnit` (not a mock) and asserts the transfer list sums to zero for fractional inputs such as `0.1` + `0.2`.

> Note: the same pattern exists in `normaliseTransferFungibleTokenWithAllowance` (line ~687, `toBaseUnit(-totalAmount, ...)`), which has the identical defect and is worth fixing together.

File: packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts [hunk 2]
```diff
@@ -615,6 +617,63 @@ export default class HederaParameterNormaliser {
...
+    const tokenInfo = await mirrornode.getTokenInfo(parsedParams.tokenId);
+    const tokenDecimals = Number(tokenInfo.decimals);
+    // Fallback to 0 if decimals are missing or NaN
+    const safeDecimals = Number.isFinite(tokenDecimals) ? tokenDecimals : 0;
+
+    const tokenTransfers: TokenTransferMinimalParams[] = [];
+    let totalBase = 0;
+
+    for (const transfer of parsedParams.transfers) {
+      const base = toBaseUnit(transfer.amount, safeDecimals).toNumber();
```
Comment (kierzniak) [inline packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts]: Maybe we should use toBaseUnitLong(transfer.amount, safeDecimals, 'Transfer amount') here, like every sibling? .toNumber() drops to float64, so we lose the int64 guard #1020 just added

File: docs/HEDERATOOLS.md [hunk 3]
```diff
@@ -836,6 +837,41 @@ Remove allowance for spender 0.0.123 on tokens 0.0.456 and 0.0.789
 
 ---
 
+### TRANSFER_FUNGIBLE_TOKEN_TOOL
+
+**Supports Hooks & Policies**: ✅ Yes
```
Comment (kierzniak) [inline docs/HEDERATOOLS.md:842]: Not true yet for policies. `TRANSFER_FUNGIBLE_TOKEN_TOOL` is missing from both `relevantTools (max-recipients-policy.ts:50)` and `_getBuiltinStrategies()` `(:138)`, so `MaxRecipientsPolicy` never inspects it and the unbounded `transfers` array has no cap. Adding it via `additionalTools` throws `Unhandled tool` at `:66`.

Params use tokenTransfers with a negative sender entry, same shape as the allowance sibling, so `_countTokenAllowance` works as-is: rename it `_countTokenTransfers` and map both tools. Plus a test, 6 recipients with `maxRecipients: 5` should block.

**Review by kierzniak — CHANGES_REQUESTED** [review]

File: packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts (hunk 2 above)
Comment (skurzyp-blockydevs) [reply packages/core/src/shared/hedera-utils/hedera-parameter-normaliser.ts]: You are right, thanks

File: docs/HEDERATOOLS.md (hunk 3 above)
Comment (skurzyp-blockydevs) [reply docs/HEDERATOOLS.md:842]: Oh, good catch. Forgot to update those. Thanks!

**Review by kierzniak — APPROVED** [review]
